src/components/Chat.js

```diff
@@ -0,0 +1,214 @@
+import 'moment/locale/de';
+import PropTypes from 'prop-types';
+import React, { useCallback, useEffect, useState } from 'react';
+import { StyleSheet, View } from 'react-native';
+import { Avatar } from 'react-native-elements';
+import {
+  Bubble,
+  Composer,
+  GiftedChat,
+  InputToolbar,
+  MessageText,
+  Send
+} from 'react-native-gifted-chat';
+
+import { colors, Icon, normalize } from '../config';
+import { momentFormat } from '../helpers';
+
+import { RegularText } from './Text';
+
+/* eslint-disable react-native/no-inline-styles */
+/* eslint-disable react/prop-types */
+
+const UserAvatar = ({ uri, title }) => (
+  <Avatar
+    containerStyle={styles.spacing}
+    overlayContainerStyle={[styles.overlayContainerStyle, !uri && styles.border]}
+    placeholderStyle={styles.placeholderStyle}
+    rounded
+    source={uri ? { uri } : undefined}
+    renderPlaceholderContent={
+      <Avatar
+        containerStyle={[styles.containerStyle]}
+        overlayContainerStyle={[styles.overlayContainerStyle, styles.border]}
+        rounded
+        title={title}
+        titleStyle={styles.titleStyle}
+      />
+    }
+  />
+);
+
+/**
+ * it is the component used to realise the chat function
+ * @param {array} data please make sure that the data format is as shown in the document
+ *                      https://github.com/FaridSafi/react-native-gifted-chat#message-object
+ * @param {object} bubbleWrapperStyleLeft  style of chat balloons on the left
+ * @param {object} bubbleWrapperStyleRight style of chat balloons on the right
+ * @param {object} messageTextStyleLeft    style of chat text on the left
+ * @param {object} messageTextStyleRight   style of chat text on the right
+ * @param {func}   onSendButton            function returning message text
+ * @param {string} placeholder             placeholder text of `textInput`
+ * @param {object} textInputProps          props to customise text input
+ * @param {number} userId                  prop to recognise whether the message is the owner
+ *                                         or another user
+ */
+export const Chat = ({
+  bubbleWrapperStyleLeft,
+  bubbleWrapperStyleRight,
+  data,
+  messageTextStyleLeft,
+  messageTextStyleRight,
+  onSendButton,
+  placeholder = 'Type a message',
+  textInputProps,
+  userId = 1
+}) => {
+  const [messages, setMessages] = useState(data);
+
+  useEffect(() => {
+    setMessages(data);
+  }, [data]);
+
+  const onSend = useCallback((messages) => {
+    onSendButton(messages[0].text);
+  }, []);
+
+  return (
+    <GiftedChat
+      alwaysShowSend
+      messages={messages}
+      minInputToolbarHeight={normalize(96)}
+      onSend={(messages) => onSend(messages)}
```

Comment:
can be shortened to `onSend={onSend}`

Reply:
or even different with

```
onSend={(messages) => {
  onSendButton(messages[0].text);
}}
```

why did you decide to have a `useCallback` without dependency? i guess we cannot profit from that.